app/src/main/java/nerd/tuxmobil/fahrplan/congress/utils/FahrplanMisc.kt

```diff
@@ -0,0 +1,66 @@
+package nerd.tuxmobil.fahrplan.congress.utils
+
+import android.app.AlarmManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+
+import info.metadude.android.eventfahrplan.commons.temporal.Moment
+import nerd.tuxmobil.fahrplan.congress.MyApp
+import nerd.tuxmobil.fahrplan.congress.alarms.AlarmReceiver
+import nerd.tuxmobil.fahrplan.congress.alarms.AlarmUpdater
+import nerd.tuxmobil.fahrplan.congress.extensions.*
+import nerd.tuxmobil.fahrplan.congress.models.DateInfo
+import nerd.tuxmobil.fahrplan.congress.models.DateInfos
+
+
+object FahrplanMisc {
+
+    private const val LOG_TAG = "FahrplanMisc"
+
+    /**
+     * Returns a [DateInfos] instance composed from the given [dateInfos] list.
+     * Duplicate [DateInfo] entries are removed.
+     */
+    fun createDateInfos(dateInfos: List<DateInfo>): DateInfos {
+        val infos = DateInfos()
+
+        for (dateInfo in dateInfos) {
+            if (!infos.contains(dateInfo)) {
```

Comment:
`!infos.contains(dateInfo)` can become `dateInfo !in infos`

Reply:
should be done in newest commit

Reply:
:x: Be careful - you mixed up the lines.

Reply:
🤦 Fixed now